src/nodes.coffee

```diff
@@ -2546,18 +2640,11 @@ exports.Assign = class Assign extends Base
   compileNode: (o) ->
     isValue = @variable instanceof Value
     if isValue
-      # When compiling `@variable`, remember if it is part of a function parameter.
-      @variable.param = @param
```

Comment:
I don't know exactly when I removed this. But went back through and it appears to make good sense - `.param` is only used on `Assign`s, so here setting `@variable.param` wouldn't have an effect since `@variable` wouldn't be an `Assign`

So this is a cleanup of unused LHS-propagation